client/heartbeatstop.go

```diff
@@ -0,0 +1,149 @@
+package client
+
+import (
+	"fmt"
+	"time"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/state"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+type heartbeatStop struct {
+	lastOk      time.Time
+	allocs      map[string]time.Duration
+	allocHookCh chan *structs.Allocation
+	getRunner   func(string) (AllocRunner, error)
+	logger      hclog.InterceptLogger
+	state       state.StateDB
+}
+
+func newHeartbeatStop(
+	state state.StateDB,
+	getRunner func(string) (AllocRunner, error),
+	logger hclog.InterceptLogger) *heartbeatStop {
+
+	h := &heartbeatStop{
+		allocs:      make(map[string]time.Duration),
```

Comment:
The name `allocs` feels a bit ambiguous. This is a map of allocation IDs to... their configured heartbeat interval?

Reply:
fixed